sklearn/naive_bayes.py

```diff
@@ -165,6 +176,12 @@ def __init__(self, priors=None, var_smoothing=1e-9):
         self.priors = priors
         self.var_smoothing = var_smoothing
 
+    def _check_input(self, X, Y=None):
```

Comment:
Could you clarify a bit what you mean here?

Reply:
I think @jnothman doesn't like the fact that your '_check_input' combines X as well as Y=None as default argument. Keep it better separated as he suggests.

Reply:
Ah, thanks for the clarification @FlorianWilhelm . Here are my thoughts on the matter, which, of course, are only my opinion:
Input checking is required for the following four functions:
- `predict`: Is implemented in `BaseNB`
- `predict_log_proba`:  Is implemented in `BaseNB`
- `fit`: Is implemented by every function
- `partial_fit`: Is implemented by every function
From an intuitive point of view, I would expect that when one of the above functions is called, first, the input is checked and then computations on the input are done. Input checking was at different places in the code and sometimes applied more than once. That is, input checking was usually done at the beginning of `_joint_likelihood` (which is called in `predict`) and the `_count` function (which is called by `fit` and `partial_fit`), as well as in the beginning of the `fit` and `partial_fit` methods. That's one of the reasons I introduced `_check_input`.
The other is that `BernoulliNB` and especially `CategoricalNB` need a more complicated input checking (also because of old numpy support). Besides `check_array` and `check_X_y` other things have to be checked too. 

I thought it was straight forward, to apply a function that can be manually configured by every class and is responsible for input checking. This has the advantage, that one has to look in one place only to understand the input checking and not at four or up to six places. Further, this function can elegantly be called in the parent routines in `BaseNB`.

If we change it, we could do the input checking (as it was before), at the beginning of `_joint_log_likelihood` which then does two things: calculate the joint likelihood and check the input (which is against the paradigm that every function should do one thing). Further, we would need to overwrite (which is actually just copying) the `fit` and `partial_fit` functions for `BernoulliNB`, just to implement a slightly different input check. To avoid this, we could implement the input checks in `_count`. But then we have the same problem as for `_joint_log_likelihood`.

To me, although a bit ugly, it seems more useful to just have to look into one place to understand the input checking and to have the input checks at the beginning of each public function, that requires input.
But, maybe my priorities are different than yours. So please tell me and I change it.